cfme/tests/v2v/test_csv_import.py

```diff
@@ -16,44 +15,48 @@
 
 pytestmark = [
     pytest.mark.provider(
-        classes=[RHEVMProvider],
-        selector=ONE_PER_VERSION,
-        scope="module"
+        classes=[RHEVMProvider], selector=ONE_PER_VERSION, required_flags=["v2v"], scope="module"
     ),
     pytest.mark.provider(
         classes=[VMwareProvider],
-        selector=ONE_PER_TYPE,
-        fixture_name='source_provider',
-        scope="module"
-    )
+        selector=ONE_PER_VERSION,
+        fixture_name="source_provider",
+        required_flags=["v2v"],
+        scope="module",
+    ),
+    pytest.mark.usefixtures("v2v_provider_setup")
 ]
 
 
 @pytest.fixture(scope="function")
-def infra_map(appliance, v2v_providers):
+def infra_map(appliance, source_provider, provider):
     """Fixture to create infrastructure mapping"""
-    form_data = _form_data(v2v_providers.vmware_provider, v2v_providers.rhv_provider)
-    return appliance.collections.v2v_mappings.create(form_data)
+    infra_mapping_data = infra_mapping_default_data(
+        source_provider, provider)
+    infrastructure_mapping_collection = appliance.collections.v2v_infra_mappings
+    mapping = infrastructure_mapping_collection.create(**infra_mapping_data)
+    yield mapping
+    infrastructure_mapping_collection.delete(mapping)
 
 
 def migration_plan(appliance, infra_map, csv=False):
     """Function to create migration plan and select csv import option"""
-    plan_name = "map_{}".format(fauxfactory.gen_alpha(10))
-    plan_obj = appliance.collections.v2v_plans
+    radio_btn = "Import a CSV file with a list of VMs to be migrated"
+    plan_obj = appliance.collections.v2v_migration_plans
     view = navigate_to(plan_obj, 'Add')
     view.general.fill({
-        'infra_map': infra_map.name,
-        'name': plan_name,
-        'description': fauxfactory.gen_alpha(20)
+        "infra_map": infra_map.name,
+        "name": fauxfactory.gen_alpha(10),
+        "description": fauxfactory.gen_alpha(10),
+        "select_vm": radio_btn
```

Comment:
Q: Would it be possible to have the import be a separate widget from `select_vm`? It seems odd to fill "select_vm" with radio_btn

Reply:
Changed it to choose_vm . They are two radio buttons to choose which way the Vm's will be imported .